File: leet/0141-linked-list-cycle/linked-list-cycle.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func hasCycle(head *ListNode) bool {
    if head == nil || head.Next == nil {return false}
    
    fast := head
    slow := head
    
    for fast != nil && slow != nil {
        fast = fast.Next 
        if fast != nil {fast = fast.Next}
        slow = slow.Next
        
        if fast == slow {
            return true
        }
    }
    
    return false
}

File: leet/0141-linked-list-cycle/linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    if (!head || !head.next) return false

    let fast = head.next.next
    let slow = head.next

    while (fast && slow){
        if(fast == slow) return true
        slow = slow.next
        fast = fast.next ? fast.next.next : null
    }
    return false
};
